pkg/source/examples/simple_source/main.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"context"
+
+	sourcesdk "github.com/numaproj/numaflow-go/pkg/source"
+	"github.com/numaproj/numaflow-go/pkg/source/examples/simple_source/impl"
+	"github.com/numaproj/numaflow-go/pkg/source/server"
+)
+
+func main() {
+	simpleSource := impl.NewSimpleSource(10)
+	server.New(
+		sourcesdk.PendingFunc(simpleSource.Pending),
+		sourcesdk.ReadFunc(simpleSource.Read),
+		sourcesdk.AckFunc(simpleSource.Ack)).
```

Comment:
```suggestion
		simpleSource).
```

here the `simpleSource` implements the source interface.